### DISCUSSION TITLE
Microsoft Defender has detected this file as a virus

### DISCUSSION THREAD

**Discussion by AoiNiCo** [34736384/genshin-fps-unlock#296] [General]
Microsoft Defender has detected this file as a virus
![image](https://github.com/34736384/genshin-fps-unlock/assets/52157900/1132791b-0e17-46ee-92cf-67fa09485d15)

**Comment by rhonivanpareja** [comment]
Yeah i have recieved the same notification on my antivirus, can someone review the changes why the file is being detected?

**Comment by Jonyee04** [comment]
it seems i am not alone, i got the same virus type, strange because yesterday i played without any warning

**Comment by Jonyee04** [comment]
i already removed because of security 
![imagem_2023-12-21_140016530](https://github.com/34736384/genshin-fps-unlock/assets/103516099/6b534126-b725-4694-948d-e322e010bfa7)

**Comment by Jonyee04** [comment]
also, we got a response for the file creator earlier on his own post
![imagem_2023-12-21_140343279](https://github.com/34736384/genshin-fps-unlock/assets/103516099/cacc8ffe-cd87-46fd-a276-cb96f0d1e048)

So yeah, if you guys don't want to use the file anyways, we need to stick at 60 fps for now

**Comment by tuanminh26596** [comment]
I know the Developer has put a lot of effort and dedication into maintaining and updating the software until now, I'm really appreciate for that. But the current version is having a lot of problems with Windows Defender, especially when it classifies this software into many different types of viruses.
![image](https://github.com/34736384/genshin-fps-unlock/assets/60099719/30448ed6-7b78-4170-8978-0f07c9a9f222)

**Comment by Jonyee04** [comment]
@tuanminh26596 yeah i appreciate it too, it's nice to have someone like him supporting the comunity, but the only reason i don't let any file that gets a warning in my system it's because i suffered a lot with trojans back then, i think it's a valid point

**Comment by Clavus** [comment]
I've tried building the project myself and Windows Defender still tripped over the executable Visual Studio spat out. So yeah, it's some virus scanner heuristic that's resulting in a false positive.

I'm not much of a C++ dev myself otherwise it'd be interesting to dive in to see if there's a workaround to avoid the trigger.

**Comment by tuanminh26596** [comment]
> @tuanminh26596 yeah i appreciate it too, it's nice to have someone like him supporting the comunity, but the only reason i don't let any file that gets a warning in my system it's because i suffered a lot with trojans back then, i think it's a valid point

Have to play Genshin at 60 fps on a 2k 164hz monitor is a torture [💀](https://emojipedia.org/skull)

**Comment by Vindfury** [comment]
> I've tried building the project myself and Windows Defender still tripped over the executable Visual Studio spat out. So yeah, it's some virus scanner heuristic that's resulting in a false positive.
> 
> I'm not much of a C++ dev myself otherwise it'd be interesting to dive in to see if there's a workaround to avoid the trigger.

I also built it myself and it worked for some reason. Microsoft isn't detecting it.
I did run into this warning `Unmanaged.cpp(72,5): warning C4267: 'argument': conversion from 'size_t' to 'DWORD', possible loss of data`
Other than that it's fine. I'm not a dev at all, but from what I can see, it looks like it depends on your compiler. Mine isn't very updated. I'm currently using Microsoft Visual Studio Community 2022 RC (64-bit) Version 17.0.0 RC3

**Comment by Andrejob99** [comment]
yes, please fix

**Comment by EdwinStoteler** [comment]
Looking at the PR to 4.3 update I expect this line in the vcxproj file to be the issue:
`<UACExecutionLevel>RequireAdministrator</UACExecutionLevel>`

At least I don't see any other changes that the program wasn't doing already before. Perhaps someone could make a branch and test it. Make a PR if it works. I don't have the tools installed on my PC (not going to use my work laptop for this).

**Reply by 34736384** [reply]
No, that's for the debug profile. It is needed for debugging
RequireAdministrator has been set in the release profile since the repo was created.
The release profile generates the binary that is uploaded over at release

**Comment by MenonPlayz** [comment]
~~The downloadable .exe doesnt get detected by VirusTotal, neither by Malwarebytes.~~ 

EDIT: I only sent the link to Virustotal, which doesnt work properly, as @Vindfury pointed out. My bad

> I also built it myself and it worked for some reason. Microsoft isn't detecting it. 

I tried the same, downloaded Visual Studio 2022, built the source code and scanned it with Windows Defender and Malwarebytes. No detection. Am using the self-built now

**Comment by Perceval21** [comment]
Porbably the problem is that Win Def is really picky.

**Comment by Vindfury** [comment]
> The downloadable .exe doesnt get detected by VirusTotal, neither by Malwarebytes. ![grafik](https://private-user-images.githubusercontent.com/90968284/292295001-4a802368-8efc-409c-98b8-7f7b07260c62.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDMxODU5NzUsIm5iZiI6MTcwMzE4NTY3NSwicGF0aCI6Ii85MDk2ODI4NC8yOTIyOTUwMDEtNGE4MDIzNjgtOGVmYy00MDljLTk4YjgtN2Y3YjA3MjYwYzYyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMjElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjIxVDE5MDc1NVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTJlYzZmNjU0YjVlZDQwNThmOTc2ZTRhMGM0NDM0YmFhNjU2N2EzZjEwNjEyYjgzMzY3OGE0ZWU2NGNiOTg3ZWUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.5JXwSnoNcYmpNXnicpu5KgW0za9Ko7RH6RqNzTsX3MA)
> 
> > I also built it myself and it worked for some reason. Microsoft isn't detecting it.
> 
> I tried the same, downloaded Visual Studio 2022, built the source code and scanned it with Windows Defender and Malwarebytes. No detection. Am using the self-built now but I wouldn't worry too much

Unfortunately, scanning via the download link doesn't work. You need to upload it yourself for VirusTotal to show actual results. I'm curious if you got the same 'size_t' to 'DWORD' warning when you compiled it yourself.

For reference this is what you'll get from VirusTotal if you upload the GitHub exe
https://www.virustotal.com/gui/file/40515ad6b9444753609a2c2d7c4da5855796dbdb202f0a0888f9c7e64800772c

And this is what I got when I compiled it myself
https://www.virustotal.com/gui/file/8dfd69a737650991637dcd3e38fb57200753540e07b5a3acc98783d23b802644

Maybe you can also check your build if you don't mind uploading yours to VirusTotal as well

**Comment by Vindfury** [comment]
> Looking at the PR to 4.3 update I expect this line in the vcxproj file to be the issue: `<UACExecutionLevel>RequireAdministrator</UACExecutionLevel>`
> 
> At least I don't see any other changes that the program wasn't doing already before. Perhaps someone could make a branch and test it. Make a PR if it works. I don't have the tools installed on my PC (not going to use my work laptop for this).

If I recall correctly, the detection started happening with the [v2.1.2](https://github.com/34736384/genshin-fps-unlock/releases/tag/v2.1.2) release. I stuck with v2.1.1 because it was working fine for 4.2 anyway

**Comment by MenonPlayz** [comment]
@Vindfury here is the one I built from source-code. Detections are the same as yours. I updated my previous comment
https://www.virustotal.com/gui/file/d67676d592ba760eff701b654080a325a4041b538a260319450ca736aff70316?nocache=1

Im gonna try and build the previous versions

**Comment by Vindfury** [comment]
@MenonPlayz Thank you! I appreciate it and yeah, I saw your updated post

**Comment by Rabydex556** [comment]
Hello. 7 hours ago today I received the same error. I tried to reinstall everything and I still get the Windows Defender alert. Hopefully the creator of an explanation or a patch to solve this quickly. I want to enjoy the new event at 120fps :(

**Comment by Delpirou-Maxime** [comment]
i have the same "issue" but as the dev said, it's a false positive. 

You  have the same issue with cheatengine for example (windows defender and some other Anti Virus said it's a trojan/malware) but if you download from a trust source there's no problem.

But again you can built in yourself but if i've read correctly even this trigger some Anti-Virus. 


I choose to trust the dev.

**Comment by MenonPlayz** [comment]
Okay, I built everything since the May 29 commit ["try to fix unlocker not working on first boot"](https://github.com/34736384/genshin-fps-unlock/commit/07e9c9449847cde4a5347ec343b7ef69b07cc3e3?diff=unified&w=0)

The VirusTotal links for my self compiled [May 29](https://www.virustotal.com/gui/file/3e15ec24fcee913e282d657a9994355cd721b96f2280ec8b116e86bc0c8533e0), [November 17 commit](https://www.virustotal.com/gui/file/b9b1d214cb67904e10b1e7597a57b168e1a8fd326ab32410b2da8e05878779b7), [December 6](https://www.virustotal.com/gui/file/bf61d49f33dbca2c53592011c39f368292a153b75f2a3c3718780ce9f8294bd7), [December 7](https://www.virustotal.com/gui/file/4b3e9804d724352f5d5922890ac93c297ae9cebaa61b8908d8752e31055c150d), [December 16](https://www.virustotal.com/gui/file/906b52cd50166d36aefec44af71a7637cbdf2bfd19eb3ed46a3b8ef4d75fd2d7), [December 17 - 1](https://www.virustotal.com/gui/file/8422ab1498cadb767b668f6bf01d606baa225627cb40c829496265764e069996), [December 17 - 2](https://www.virustotal.com/gui/file/30cefcb137f031d515268fb6180d51d17b2dd6df8a2fac767ba4201d1bfe36db) and [December 19 aka latest commit](https://www.virustotal.com/gui/file/e0e36bce6796a0e9d51966957d0945f7b895ed7782e3ffa734f5da1496741322) builds seem to show the same detections.
They are different from scanning the downloaded exe of the 2.0.0 release, which gives [this VirusTotal report](https://www.virustotal.com/gui/file/40515ad6b9444753609a2c2d7c4da5855796dbdb202f0a0888f9c7e64800772c).

None get detected by Microsoft Defender (both on my PC and in VirusTotal, I manually selected and scanned all of the exes).
While doing this, I realised I should check older releases as a comparison. 
[Release v1.4.2](https://www.virustotal.com/gui/file/b40b9fff8d7472e46454323df8cbbabf0b30aa07cd04b5560056e7b03da72b48) and [Release v1.0.0](https://www.virustotal.com/gui/file/f62fae37db7503ea3c98c81f87d1d87b811579872190b97f4134b33606ed853a) did show some slightly different things.

I dont know a lot about malware, am new to C++ and new to this unlocker. So I can only give my findings to y'all

**Comment by Rabydex556** [comment]
> Bien, construí todo desde la confirmación del 29 de mayo ["intente arreglar que el desbloqueo no funcione en el primer arranque"](https://github.com/34736384/genshin-fps-unlock/commit/07e9c9449847cde4a5347ec343b7ef69b07cc3e3?diff=unified&w=0)
> 
> Los enlaces de VirusTotal para mi autocompilación [del 29 de mayo](https://www.virustotal.com/gui/file/3e15ec24fcee913e282d657a9994355cd721b96f2280ec8b116e86bc0c8533e0) , [confirmación del 17 de noviembre](https://www.virustotal.com/gui/file/b9b1d214cb67904e10b1e7597a57b168e1a8fd326ab32410b2da8e05878779b7) , [6 de diciembre](https://www.virustotal.com/gui/file/bf61d49f33dbca2c53592011c39f368292a153b75f2a3c3718780ce9f8294bd7) , [7 de](https://www.virustotal.com/gui/file/4b3e9804d724352f5d5922890ac93c297ae9cebaa61b8908d8752e31055c150d) diciembre , [16 de diciembre](https://www.virustotal.com/gui/file/906b52cd50166d36aefec44af71a7637cbdf2bfd19eb3ed46a3b8ef4d75fd2d7) , [17 y 1 de diciembre](https://www.virustotal.com/gui/file/8422ab1498cadb767b668f6bf01d606baa225627cb40c829496265764e069996) , [17 y 2 de diciembre](https://www.virustotal.com/gui/file/30cefcb137f031d515268fb6180d51d17b2dd6df8a2fac767ba4201d1bfe36db) y [19 de diciembre, también conocidas como las últimas](https://www.virustotal.com/gui/file/e0e36bce6796a0e9d51966957d0945f7b895ed7782e3ffa734f5da1496741322) compilaciones de confirmación, parecen mostrar las mismas detecciones. Son diferentes de escanear el archivo ejecutable descargado de la versión 2.0.0, que proporciona [este informe de VirusTotal](https://www.virustotal.com/gui/file/40515ad6b9444753609a2c2d7c4da5855796dbdb202f0a0888f9c7e64800772c) .
> 
> Microsoft Defender no detecta ninguno (tanto en mi PC como en VirusTotal, seleccioné y escaneé manualmente todos los archivos ejecutables). Mientras hacía esto, me di cuenta de que debería comprobar las versiones anteriores a modo de comparación. [La versión v1.4.2](https://www.virustotal.com/gui/file/b40b9fff8d7472e46454323df8cbbabf0b30aa07cd04b5560056e7b03da72b48) y [la versión v1.0.0](https://www.virustotal.com/gui/file/f62fae37db7503ea3c98c81f87d1d87b811579872190b97f4134b33606ed853a) mostraron algunas cosas ligeramente diferentes.
> 
> No sé mucho sobre malware, soy nuevo en C++ y nuevo en este desbloqueador. Así que sólo puedo darles mis hallazgos a todos ustedes.

Excuse me, can you give me a small class on how to compile from Visual Studio so I can use it in official Genshin? I am 100% new to programming

**Comment by 6667ab** [comment]
I don't buy it. Why would a personally compiled version of the same exe ring up with so many less positives than the author compiled version?
It's extremely suspicious and I don't think there's a good explanation for it that doesn't involve the author being malicious somehow.
I would recommend changing your hoyoverse passwords and deleting the exe for now at the _very_ least. Once the author is willing to upload a version that has the same """false positives""" as a handcompiled one, we'll be closer to "good", but I don't know if I'd trust this github going forward.

**Reply by 34736384** [reply]
Look, I don't blame you for not having the ability to use a disassembler like IDA to compare the binary with the source code. But I don't need your smartass accusing me shit when you have no evidence that the binaries I uploaded actually contains malicious code. And if you ever have the ability to use IDA then you'll find that it is exactly the same as the source code posted.

**Reply by Clavus** [reply]
I get it that folks wouldn't want to use this application as long as it's triggering virus scanners. But just because you can't think of an explanation doesn't mean you get to throw these accusations around. We've looked at the code and reproduced the issue, so you have no business barging in and claiming the author is untrustworthy unless there's actual proof.

**Reply by Adebasquer** [reply]
For any confused about this, windows usually adds a mark of the web zone identifier in the alternate data stream of downloaded files, which windows defender (and other antiviruses) usually treats with more strict screening than files created locally (this is also why many downloaded .exe files display the "Windows Protected your PC" popup). So downloading the source code and compiling it yourself will generally result in less detections (false positive or otherwise).

**Comment by Tonic0** [comment]
Some tips for anyone else struggling with this.

I installed Visual Studio 2019 CE as instructed for compiling by the dev. I then took the 2.1.1 source (last version which didn't give any Defender trojan warnings) and implemented all the FPS detection logic from the latest 2.2.0 version, but didn't include <UACExecutionLevel>RequireAdministrator</UACExecutionLevel> in unlockfps_clr.vcxproj

This still gave me warnings if uploaded to Virustotal, but on my PC this stopped Defender from complaining. Scanning it manually doesn't give anything malicous either. I do have to run the exe manually as admin or toggle it from file properties, but so far it has worked for 3 days straight with no trojan warnings.

**Comment by lobotomy-x** [comment]
lol guys its not that he suddenly snuck malware in, the reason behavior changed was windows updates. Recently they put out some security updates that have done more harm than good. And defender has always been overtuned. I can make a new reg file, put in keys to disable windows defender from always scanning and damaging my files (i.e. malware behavior) and defender will call it a Trojan after literally witnessing its birth. 

I have defender disabled in all official settings but probably have to fully remove it as even while disabled it seems to somehow still be preventing me from hooking both this app and 3dmigoto following a security update a few days ago


Oh my god its worse than I thought. These "detections" are literally from turning settings off in the security panel itself. Please do not take windows defender seriously lol use bitdefender if you must use an active antivirus.
![image](https://github.com/34736384/genshin-fps-unlock/assets/10409982/bc37a64e-9e2f-401b-875a-4ffbf2621731)

**Reply by MenonPlayz** [reply]
@elbadcode I would NOT advise disabling windows defender. It's a good, simple, free anti virus compared to all these free ad-wares like Avast, Avira, McAfee. 
If you want to bypass the detections, just build the exe from source in Visual Studio.

@Rabydex556 
Its not too complicated. Download Visual Studio [here](https://visualstudio.microsoft.com/de/vs/features/cplusplus/) if you havent already. Download the [latest source code](https://github.com/34736384/genshin-fps-unlock/archive/refs/tags/v2.2.0.zip), extract it and open the unlockfps_clr.sln file in Visual Studio. Then you go to build at the top navigation bar and click build solution, or use the Shortcut Ctrl+Shift+B. You will find the unlockfps_clr.exe in the new x64>Debug folder. 
It should not produce the windows defender detection anymore

**Reply by Andrejob99** [reply]
Thanks a bunch, I felt really dumb not knowing how to make my own executable and nothing I found looked simple enough

**Reply by lobotomy-x** [reply]
Defender has caused me nothing but issues and never detected an actual malware. I wouldn't recommend everyone go disable it, but I'll never use it if I can help it. In any case I resolved failure to launch by moving my 3dmigoto folder and unlockfps inside my documents folder rather than C:/games where I previously put them. I never had outright detections as I had disabled every part of defender except the background service, but until I did that the game would simply not launch when I hit play despite working a few days ago

**Reply by Perceval21** [reply]
@MenonPlayz i used this solution too, thanks for the tip

**Comment by rardar1993** [comment]
> @elbadcode I would NOT advise disabling windows defender. It's a good, simple, free anti virus compared to all these free ad-wares like Avast, Avira, McAfee. If you want to bypass the detections, just build the exe from source in Visual Studio.
> 
> @Rabydex556 Its not too complicated. Download Visual Studio [here](https://visualstudio.microsoft.com/de/vs/features/cplusplus/) if you havent already. Download the [latest source code](https://github.com/34736384/genshin-fps-unlock/archive/refs/tags/v2.2.0.zip), extract it and open the unlockfps_clr.sln file in Visual Studio. Then you go to build at the top navigation bar and click build solution, or use the Shortcut Ctrl+Shift+B. You will find the unlockfps_clr.exe in the new x64>Debug folder. It should not produce the windows defender detection anymore

Thank you MenonPlayz. I followed your instructions and it worked for me. The only thing I wasn't c;lear on but found the information on within Visual Studio was where the new .exe file after Building Solution was. Like you mentioned, it was in the x64>Debug folder, and those folders were located in the original .sln file folder location.

**Comment by sakana810** [comment]
Ihave no idea about program but I can build exe thanks to your tip and GPT.
Thank you sir!

**Comment by raygiizs** [comment]


**Reply by 34736384** [reply]
Rewriting the project is not meant for detection evasion in the first place

**Reply by MenonPlayz** [reply]
recent releases dont seem to get flagged anymore, so that's good.
Btw @34736384 what do the _clr and _nc stand for on the exe names

**Reply by MenonPlayz** [reply]
Latest downloaded version didnt get detected at first. Then I play the game for a bit and Defender removes the downloaded exe.
![grafik](https://github.com/34736384/genshin-fps-unlock/assets/90968284/b491c9e8-c22b-46a7-9fae-76a1d3537d75)


Im gonna use the self-built again, though I cant get it to work without additional dlls in the same folder.

**Comment by iLStrix** [comment]
It still gets tagged on VirusTotal aswell: https://www.virustotal.com/gui/file/08f16aa2f37dcd7fb4748c360a12f17e1fef33906be4cb8586874237af6f5fec/detection

EDIT: Althought that does look a lot like false positives and went from 3 flags to 2 flags. The remaining 2 flags are both AI.
